### PR TITLE
Add create timezone name list feature API

### PR DESCRIPTION
**Pull request by tigershi**

### CODE REVIEW

File: g11n-ws/modules/md-restful-api/src/main/java/com/vmware/vip/api/rest/APIParamValue.java [hunk 1]
```diff
@@ -30,6 +30,8 @@
 	public final static String REGION         = "a string which represents region, e.g. US, PT, CN";
 	public final static String COMBINE        = "an integer which represents combine type number 1 or 2";
 	public final static String SCOPE_FILTER   = "a String for filtering the pattern data, separated by comma and underline. e.g. 'dates_eras,dates_dayPeriodsFormat'";
+	public final static String DEFAULT_TERRITORY = "a boolean which represents get the default territory timezone name or not. (default is true)";
```
Comment (linr211) [inline g11n-ws/modules/md-restful-api/src/main/java/com/vmware/vip/api/rest/APIParamValue.java:33]: a boolean value to get the default territory timezone name or not, default is true.

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/date/DateFormatService.java [hunk 2]
```diff
@@ -39,4 +64,54 @@ public String formatDate(String locale, long date, String pattern) throws L2APIE
...
+	
+	/**
+   	 * @param locale
+   	 * @param default territory
+   	 * @return matching locale TimeZoneName
+   	 */
+	@Override
+	public TimeZoneName getTimeZoneName(String oldLocale, boolean defaultTerritory) throws L2APIException {
+		// TODO Auto-generated method stub
+		String locale = oldLocale.replace("_", "-");
```
Comment (linr211) [inline g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/date/DateFormatService.java:76]: I don't see where to use the oldLocale, is it a invalid parameter?

Comment (tigershi) [reply g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/date/DateFormatService.java:76]: oldLocale.replace("_", "-"),  this variable use to record the locale string that Illegal request local string parameter and reply to
client side

Comment (tigershi) [reply g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/date/DateFormatService.java]: done

File: g11n-ws/modules/md-restful-api/src/main/java/com/vmware/vip/api/rest/APIParamValue.java (hunk 1 above)
Comment (tigershi) [reply g11n-ws/modules/md-restful-api/src/main/java/com/vmware/vip/api/rest/APIParamValue.java]: done

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/dao/pattern/PatternDaoImpl.java [hunk 3]
```diff
@@ -15,6 +17,17 @@ public String getPattern(String locale, String categories) {
 		locale = locale.replace("_", "-");
 		return PatternUtil.getPatternFromLib(locale, categories);
 	}
+	
+	
+	/**
+	 * @param locale
+	 * @param default territory
```
[the remark this answers is not in the document]
Comment (tigershi) [reply g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/dao/pattern/PatternDaoImpl.java]: done

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/dao/pattern/PatternDaoImpl.java [hunk 4]
```diff
@@ -15,6 +17,17 @@ public String getPattern(String locale, String categories) {
...
+	 * @return matching locale TimeZoneName
```
[the remark this answers is not in the document]
Comment (tigershi) [reply g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/dao/pattern/PatternDaoImpl.java]: done

**Pull request merged**